### PR TITLE
Remove ability to specify arbitrary node attributes with `node.` prefix

### LINKED ISSUE #17280: Remove ability to specify arbitrary node attributes with `node.` prefix
Today the basic node settings like `node.data` and `node.master` can't really be fully validated since we allow to specify custom user attributes on the node level. We have to, in order to support that, add a wildcard setting for `node.*` to let these setting pass validation. Instead we should require a more contraint prefix like `node.attribute.` that defines a namespace that is reserved for user attributes. 
This can be breaking since validation will kick in and provide an error message accordingly.

### PR DESCRIPTION
**Pull request by s1monw**
Today the basic node settings like `node.data` and `node.master` can't really be fully validated
since we allow to specify custom user attributes on the node level. We have to, in order to
support that, add a wildcard setting for `node.*` to let these setting pass validation.
Instead we should require a more contraint prefix like `node.attr.` that defines a namespace
that is reserved for user attributes.
This commit adds a new namespace for attributes in `node.attr`.

Closes #17280

### CODE REVIEW

**Comment by dadoonet** [comment]
I think this page needs also an update: https://github.com/elastic/elasticsearch/blob/master/docs/reference/modules/cluster/allocation_awareness.asciidoc

**Comment by dadoonet** [comment]
Should we also change anything for this line? https://github.com/elastic/elasticsearch/blob/master/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy#L136

**Comment by s1monw** [comment]
> Should we also change anything for this line? https://github.com/elastic/elasticsearch/blob/master/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy#L136

no - this is a real setting?

File: docs/reference/migration/migrate_5_0/settings.asciidoc [hunk 1]
```diff
@@ -26,6 +26,13 @@ should be used instead.
 The `name` setting has been removed and is replaced by `node.name`. Usage of
 `-Dname=some_node_name` is not supported anymore.
 
+==== Node attribute settings
+
+Node level attributes used for allocation filtering, forced awareness or other node identification / grouping
+must be prefixed with `node.attr`. In previous versions it was possible to specify node attributes with the `node.`
+prefix. All node attributes except of `node.master`, `node.data` and `node.ingest` must be moved to the new `node.attr.`
```
Comment (dadoonet) [inline docs/reference/migration/migrate_5_0/settings.asciidoc]: What about `node.mode`? Is this something we need to change? I mean update the code and this doc https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc#other-test-options?

Comment (s1monw) [inline docs/reference/migration/migrate_5_0/settings.asciidoc]: node.mode is not a node attribute but `ingest, master, data` are built-in?

**Comment by dadoonet** [comment]
> no - this is a real setting?

I have no idea... Just found this file while looking at your PR. But you wrote at some point that it was an "official" setting: https://github.com/elastic/elasticsearch/commit/9b3559f4732e1902f06ca9995a1f9027c61933bd

> `tests.portsfile` is now an official setting and has been renamed to `node.portsfile`

**Comment by dadoonet** [comment]
The change looks good to me. I'm fine with the namespace `node.attr` BTW.
May be someone else would like to review it?

File: docs/reference/migration/migrate_5_0/settings.asciidoc (hunk 1 above)
Comment (javanna) [inline docs/reference/migration/migrate_5_0/settings.asciidoc]: no this is a declared setting, see Node#NODE_MODE_SETTING

**Comment by javanna** [comment]
looks great, I think we need to update also `filtering.asciidoc`

**Comment by s1monw** [comment]
@javanna @dadoonet updated the PR

**Comment by javanna** [comment]
LGTM I am so happy this gets in!

**Pull request merged**